### PR TITLE
chore(DescriptionList): update tests

### LINKED ISSUE #9673: Description list unit tests update

### PR DESCRIPTION
**Pull request by kmcfaul**
**What**: Closes #9673

### CODE REVIEW

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx [hunk 1]
```diff
@@ -1,130 +1,207 @@
...
+});
+
+test('DescriptioinListTermHelpTextButton renders custom className', () => {
+  render(
+    <DescriptionListTermHelpText>
+      <DescriptionListTermHelpTextButton className="custom">test</DescriptionListTermHelpTextButton>
+    </DescriptionListTermHelpText>
+  );
+  expect(screen.getByText('test')).toHaveClass('custom');
+});
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx:187]:
```suggestion
test('DescriptioinListTermHelpTextButton renders custom className', () => {
  render(
      <DescriptionListTermHelpTextButton className="custom">test</DescriptionListTermHelpTextButton>
  );
  expect(screen.getByText('test')).toHaveClass('custom');
});
```

**Review by thatblindgeye — CHANGES_REQUESTED** [review]
Some initial comments below. We should also have each sub-component with its own test file, and for each we should add tests that check the default classes are rendered (currently the tests are only checking that a custom class renders)

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx [hunk 2]
```diff
@@ -1,130 +1,110 @@
 import React from 'react';
-import { render } from '@testing-library/react';
+import { render, screen } from '@testing-library/react';
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx:2]: Could we add additional tests to check that the `style` object is being applied correctly when the applicable props are possed? isHorizontal + horizontalTermWidthModifier, termWidth, and isAutoFit + autoFitMinModifier

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx [hunk 3]
```diff
@@ -1,130 +1,110 @@
...
 
-  test('2 col on all breakpoints', () => {
-    const { asFragment } = render(
-      <DescriptionList columnModifier={{ default: '2Col', md: '2Col', lg: '2Col', xl: '2Col', '2xl': '2Col' }} />
-    );
-    expect(asFragment()).toMatchSnapshot();
-  });
+test(`Renders default class ${styles.descriptionList}`, () => {
+  render(<DescriptionList aria-label="list" />);
+  expect(screen.getByLabelText('list')).toHaveClass(styles.descriptionList);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionList.test.tsx:14]: should be able to pass the exact option:

```suggestion
  expect(screen.getByLabelText('list')).toHaveClass(styles.descriptionList, {exact: true});
```

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionListDescription.test.tsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(<DescriptionListDescription>test</DescriptionListDescription>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders default class ${styles.descriptionListDescription}`, () => {
+  render(<DescriptionListDescription>test</DescriptionListDescription>);
+  expect(screen.getByText('test').parentElement).toHaveClass(styles.descriptionListDescription);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionListDescription.test.tsx:14]:
```suggestion
  expect(screen.getByText('test').parentElement).toHaveClass(styles.descriptionListDescription, {exact: true});
```

Also a test that spread props works correctly

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionListGroup.test.tsx [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(<DescriptionListGroup>test</DescriptionListGroup>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders default class ${styles.descriptionListGroup}`, () => {
+  render(<DescriptionListGroup>test</DescriptionListGroup>);
+  expect(screen.getByText('test')).toHaveClass(styles.descriptionListGroup);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionListGroup.test.tsx:14]:
```suggestion
  expect(screen.getByText('test')).toHaveClass(styles.descriptionListGroup, {exact: true});
```

Also a test that spread props works correclty

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionListHelpText.test.tsx [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(<DescriptionListTermHelpText>test</DescriptionListTermHelpText>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders default class ${styles.descriptionListTerm}`, () => {
+  render(<DescriptionListTermHelpText>test</DescriptionListTermHelpText>);
+  expect(screen.getByText('test')).toHaveClass(styles.descriptionListTerm);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionListHelpText.test.tsx:14]:
```suggestion
  expect(screen.getByText('test')).toHaveClass(styles.descriptionListTerm, {exact: true});
```

Also a test to check props spread correctly. This file name should also be updated to `DescriptionListTermHelpText`

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionListHelpTextButton.test.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(<DescriptionListTermHelpTextButton>test</DescriptionListTermHelpTextButton>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders default class ${styles.descriptionListText}`, () => {
+  render(<DescriptionListTermHelpTextButton>test</DescriptionListTermHelpTextButton>);
+  expect(screen.getByText('test')).toHaveClass(styles.descriptionListText);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionListHelpTextButton.test.tsx:14]:
```suggestion
  expect(screen.getByText('test')).toHaveClass(styles.descriptionListText, {exact: true});
```

Also a test that props are spread correctly. File name should also be updated to `DescriptionListTermHelpTextButton`

File: packages/react-core/src/components/DescriptionList/__tests__/DescriptionListTerm.test.tsx [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+import styles from '@patternfly/react-styles/css/components/DescriptionList/description-list';
+
+test('Renders to match snapshot', () => {
+  const { asFragment } = render(<DescriptionListTerm>test</DescriptionListTerm>);
+  expect(asFragment()).toMatchSnapshot();
+});
+
+test(`Renders default class ${styles.descriptionListTerm}`, () => {
+  render(<DescriptionListTerm>test</DescriptionListTerm>);
+  expect(screen.getByText('test').parentElement).toHaveClass(styles.descriptionListTerm);
```
Comment (thatblindgeye) [inline packages/react-core/src/components/DescriptionList/__tests__/DescriptionListTerm.test.tsx:14]:
```suggestion
  expect(screen.getByText('test').parentElement).toHaveClass(styles.descriptionListTerm, {exact: true});
```

Also a test for checking props spread correctly

**Review by thatblindgeye — CHANGES_REQUESTED** [review]

**Review by thatblindgeye — APPROVED** [review]

**Review by wise-king-sullyman — APPROVED** [review]

**Pull request merged**